File: CodeWars/7kyu/Find_the_index_of_the_second_occurrence_of_a_letter_in_a_string.py
# In this kata, you need to write a function that takes a string and a letter as input and then returns the index of the second occurrence of that letter in the string. If there is no such letter in the string, then the function should return -1. If the letter occurs only once in the string, then -1 should also be returned.
#
# Examples:
#
# second_symbol('Hello world!!!','l') --> 3
# second_symbol('Hello world!!!', 'A') --> -1
# Good luck ;) And don't forget to rate this Kata if you liked it.
#
# FUNDAMENTALSSTRINGS
# Solution
def second_symbol(s, sym):
    return [i for i in range(len(s)) if s[i] == sym][1] if sym in s and s.count(sym) > 1 else -1

File: CodeWars/7kyu/Find_the_index_of_the_second_occurrence_of_a_letter_in_a_string.cpp
/*
In this kata, you need to write a function that takes a string and a letter as input and then returns the index of the second occurrence of that letter in the string. If there is no such letter in the string, or if the letter occurs only once in the string, -1 should be returned.

Examples:

for inputs  "Hello world!!!", 'l'  ->  return 3
for inputs  "Hello world!!!", 'A'  ->  return -1
Good luck ;) And don't forget to rate this kata if you liked it.

FundamentalsStrings
*/
// Solution
#include <string>

int secondSymbol(const std::string& str, char symbol) {
  bool isSeen = false;
  for (int idx = 0; idx < str.size(); ++idx) {
    if (str[idx] == symbol) {
      if (isSeen) {
        return idx;
      }
      isSeen = true;
    }
  }
  return -1;
}